providers/azureblob/src/test/java/org/jclouds/azure/storage/filters/SharedKeyLiteAuthenticationTest.java

```diff
@@ -74,6 +74,23 @@ void testIdempotent(HttpRequest request) {
       System.out.printf("%s: %d iterations before the timestamp updated %n", Thread.currentThread().getName(),
             iterations);
    }
+   
+   @Test 
```

Comment:
These unit tests cover a tiny part of the functionality.  We need tests like `apis/s3/src/test/java/org/jclouds/s3/filters/RequestAuthorizeSignatureV2Test.java` and `apis/s3/src/test/java/org/jclouds/s3/S3ClientLiveTest.java` to ensure this functionality works.

Reply:
Yes. Take the idempotent test as an example. You may probably want to be able to create a filter with a SAS token credential and check that the filtered request has the expected content.

Also, you need to add a new `AzureBlobHttpApiModuleTest.java` and add proper unit tests for the `authSAS` method with different credential formats as an input, to make sure it produces the right result for all variety of valid and invalid inputs.

Reply:
I added a new test to SharedKeyLiteAuthenticationTest, which is somehow similar to testIdempotent, which checks request filtering when it comes to SAS.